### PR TITLE
test: simplify go-test

### PR DESCRIPTION
**Pull request by satoru-takeuchi**
Remove duplicated description between the default configuration and the custom namespace configuration.

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [x] Documentation has been updated, if necessary.
- [x] Unit tests have been added, if necessary.
- [x] Integration tests have been added, if necessary.

### CODE REVIEW

File: .github/workflows/go-test-config/action.yaml [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    required: false
+
+runs:
+  using: "composite"
+  steps:
+    - name: set environment variables
+      run: |
+        OP_NS_OPT=""
+        CLUSTER_NS_OPT=""
+        test -n "${{ input.op-ns }}" && OP_NS_OPT="--operator-namespace ${{ input.op-ns }}"
```
Comment (subhamkrai) [inline .github/workflows/go-test-config/action.yaml:18]:
```suggestion
        test -n "${{ input.op-ns }}" && OP_NS_OPT="--operator-namespace ${{ input.op-ns }}"
```

`test -n` here is for?

**Review by subhamkrai — COMMENTED** [review]
also CI is not running please check that also @satoru-takeuchi thanks

File: .github/workflows/go-test-config/action.yaml (hunk 1 above)
Comment (satoru-takeuchi) [reply .github/workflows/go-test-config/action.yaml:18]: It's old code. OP_NS_OPT should be set iff input.op-ns isn't rook-ceph. CLUSTER_NS_OPT should also be set iff input.cluster-ns isn't rook-ceph. I'll fix them.

File: .github/workflows/cluster-setup/action.yaml [hunk 2]
```diff
@@ -31,10 +31,10 @@ runs:
...
     - name: deploy rook cluster
       shell: bash --noprofile --norc -eo pipefail -x {0}
-      if: inputs.op-ns == '' || inputs.cluster-ns == ''
+      if: inputs.op-ns == 'rook-ceph' || inputs.cluster-ns == 'rook-ceph'
       run: tests/github-action-helper.sh deploy_rook
 
     - name: deploy rook cluster in custom namespace
       shell: bash --noprofile --norc -eo pipefail -x {0}
-      if: inputs.op-ns != '' || inputs.cluster-ns != ''
+      if: inputs.op-ns != 'rook-ceph' || inputs.cluster-ns != 'rook-ceph'
```
Comment (subhamkrai) [inline .github/workflows/cluster-setup/action.yaml:39]:
```suggestion
      if: inputs.op-ns != '=test-operator' || inputs.cluster-ns != '=test-cluster'
```

**Review by subhamkrai — CHANGES_REQUESTED** [review]
we want of the CI runs to run on different namespace than `rook-ceph`

File: .github/workflows/cluster-setup/action.yaml (hunk 2 above)
Comment (satoru-takeuchi) [reply .github/workflows/cluster-setup/action.yaml:39]: You meant `if: inputs.op-ns == 'test-operator' || inputs.cluster-ns == 'test-cluster'`?

Comment (subhamkrai) [reply .github/workflows/cluster-setup/action.yaml:39]: yes, typo above

**Review by subhamkrai — APPROVED** [review]
lgtm

**Pull request merged**